src/Aspire.Hosting.Azure/Extensions/AzureWebPubSubResource.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding the Azure Web PubSub resources to the application model.
+/// </summary>
+public static class AzureWebPubSubExtensions
+{
+    /// <summary>
+    /// Adds an Azure Web PubSub resource to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureWebPubSubResource> AddAzureWebPubSub(this IDistributedApplicationBuilder builder, string name)
+    {
+        var resource = new AzureWebPubSubResource(name);
+
+        return builder.AddResource(resource)
+                      .WithParameter("name", resource.CreateBicepResourceName())
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
+                      // TODO: add hub settings
```

Comment:
What's the plan to address this TODO?

Reply:
Hub settings are used for Azure Web PubSub to send events to the web app, (kind of similar to when event grid sends out events), here contains some detailed description: https://learn.microsoft.com/en-us/azure/azure-web-pubsub/howto-develop-eventhandler 

For now when we not yet support hub setting configuration, the basic workflow (as shown in the playground) works, so I think supporting hub settings is not a blocking feature. I tend to get AddAzureWebPubSub available to aspire first, how do you think?

An ideal workflow to enable configuring hub settings would be:
* When running locally, hub settings are set as "tunnel:///" with awps-tunnel container running as a sidecar (this tool as described here https://learn.microsoft.com/en-us/azure/azure-web-pubsub/howto-web-pubsub-tunnel-tool?tabs=bash is for local development), as similar to Azure Storage Emulator but yet both the container and WebPubSub service are needed) 
* When `azd up` deploying to cloud, aspire is able to add a hook to update the hub settings to the deployed website URL. As my last sync with @davidfowl this feature is on the way.

Reply:
> When azd up deploying to cloud, aspire is able to add a hook to update the hub settings to the deployed website URL. As my last sync with @davidfowl this feature is on the way.

I'm not sure we'll be able to do anything here for GA. We're still attempting to model this.